core/hw/pvr/Renderer_if.cpp

```diff
@@ -201,9 +201,9 @@ bool rend_single_frame(const bool& enabled)
 			return false;
 		if (do_swap)
 		{
-			do_swap = false;
 			if (renderer->Present())
 			{
+                do_swap = false;
```

Comment:
This would affect all platforms

Reply:
Is this really necessary? It's not quite correct since do_swap should be reset, whether or not there's a frame to present.

Reply:
This is where I don’t understand, but this is how Soul Calibur can run at 60fps on Mac now, may be adding a `__APPLE__` ?

Reply:
I need to investigate deeper then. May be Soul Calibur is doing something funky behind the scene.

Reply:
It was mentioned on discord that this happens on Android too with Soul Calibur.

Reply:
So yes, SC isn't using the framebuffer registers in the expected way, failing every other fail swap detection and effectively reducing the frame rate by half.

And this happens on all platforms. So I'll work on a proper fix and this change isn't necessary.

Reply:
thanks for investigating, reverted to original now